hs/src/Daml/Cucumber.hs

```diff
@@ -1,113 +1,144 @@
-module Daml.Cucumber where
+module Daml.Cucumber
+  ( runTestSuite
+  ) where
 
-import System.Exit
-import Data.Char
-import Control.Monad.Extra
+import Control.Monad
 import Control.Monad.IO.Class
 import Control.Monad.State (State, modify, runState, runStateT)
 import Daml.Cucumber.Daml.Parse
 import Daml.Cucumber.LSP
 import Daml.Cucumber.Parse
 import Daml.Cucumber.Types
-import qualified Data.Aeson as Aeson
-import qualified Data.ByteString.Lazy as LBS
+import Data.Char
 import Data.Foldable
 import Data.Map (Map)
-import qualified Data.Map as Map
-import qualified Data.Set as Set
+import Data.Map qualified as Map
+import Data.Set qualified as Set
 import Data.Set (Set)
-import qualified Data.Text as T
+import Data.Text qualified as T
 import Data.Text (Text)
-import qualified Data.Text.IO as T
+import Data.Text.IO qualified as T
 import Data.Traversable
 import Reflex
-import System.Directory
+import System.Directory.Contents qualified as Dir
+import System.Exit
 import System.FilePath hiding (hasExtension)
 import System.IO
 import Text.Casing
+import Text.Parsec.Error qualified as Parsec
+import Text.Parsec.Pos qualified as Parsec
 
-data Opts = Opts
-  { _opts_directory :: FilePath
-  , _opts_featureFile :: Maybe FilePath
-  , _opts_damlSourceDir :: FilePath
-  , _opts_allowMissing :: Bool
-  }
+-- | Check whether a filepath has a particular extension (including ".")
+hasExtension :: String -> FilePath -> Bool
+hasExtension ext path = takeExtension path == ext
```

Comment:
this picks up `.daml` the previous implementation implied this was undesirable. does this do the right thing?

Reply:
@madeline-os @ali-abrar It does do the right thing, in that it generates the file from only files that are needed, but we should still be disregarding .daml as nothing in there is needed for generation.

Reply:
I guess on a more fundamental note, the file .daml _doesn't_ actually have the extension .daml, which is why I wrote it the way it was. As a dot file is not recognized to have an extension in the way dot files are used.